src/web-ui/src/App.js

```diff
@@ -68,119 +58,119 @@ class App extends Component {
-                    />
-                  </Col>
-                </Row>
-                <Row style={{ marginTop: "20px" }}>
-                  <Col md={4} sm={6}>
-                    <h3>Trends for last hour</h3>
-                    <PolarChart
-                      data={Object.keys(this.state.aggregate).map(
-                        sentiment => ({
-                          x: sentiment,
-                          y: this.state.aggregate[sentiment]
-                        })
-                      )}
-                    />
-                  </Col>
-                  <Col md={4} sm={6}>
-                    <h3 style={{ marginBottom: "40px" }}>Engagement Meter</h3>
-                    <HalfCircleMeter
-                      backgroundColor="#fff"
-                      foregroundColor="#FF9900"
-                      value={this.state.happyometer}
-                      size={250}
-                    />
-                  </Col>
-                </Row>
-              </Grid>
-            </Col>
-          </Row>
-        </Grid>
-      </div>
-    );
-  }
-}
-
-export default App;
+  };
+
+  const toggleRekognition = () => {
+    const newIterationState = !iterating;
+    setIterating(newIterationState);
+
+    if (newIterationState) {
+      setIterationCompleted(false);
+      gateway.getPeople().then(response => {
+        setPeople(response.people);
+        getSnapshot();
+      });
+    }
+  };
+
+  useEffect(() => {
+    if (iterating && iterationCompleted) {
+      setIterationCompleted(false);
+      setTimeout(getSnapshot, 300);
+    }
+  }, [iterating, iterationCompleted]);
+
+  return (
+    <div className="App">
+      <Header
+        toggleRekognition={toggleRekognition}
+        addUser={addUser}
+        readyToStream={readyToStream}
+      />
+      <Grid>
+        <CameraHelp show={!readyToStream} />
+        <Row>
+          <Col md={8} sm={6}>
+            <Grid>
+              <Row>
+                <Col md={8} sm={6}>
+                  <Webcam
+                    ref={setupWebcam}
+                    screenshotFormat="image/jpeg"
+                    videoConstraints={{
+                      width: 1280,
+                      height: 640,
+                      facingMode: "user"
+                    }}
+                    width="100%"
+                    height="100%"
+                  />
+                </Col>
+                <Col md={4} sm={6}>
+                  <EngagementSummary
+                    detectedFaces={detectedFaces}
+                    detectedPeople={detectedPeople}
+                    webcamCoordinates={webcamCoordinates}
+                  />
+                </Col>
+              </Row>
+              <Row style={{ marginTop: "20px" }}>
+                <Col md={4} sm={6}>
+                  <h3>Trends for last hour</h3>
+                  <PolarChart
+                    data={Object.keys(aggregate).map(sentiment => ({
```

Comment:
For such things you can use useMemo, very convenient, maybe you don't have performance issue, but it feels good and is good to start experimenting
```
const wathever = useMemo(Object.keys(aggregate).map(...), [aggregate])
```

Reply:
In this case, `aggregate` is just an array of 5 items with `key: Number` so I think it's fine. But thanks for the comment, I'll definitely play with `useMemo` in other scenarios.